mne/viz/topomap.py

```diff
@@ -1578,11 +1572,268 @@ def _prepare_topomap(pos, ax):
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        The figure.
+    anim : instance of matplotlib FuncAnimation
+        Animation of the topomap.
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib import animation
+    if ch_type is None:
+        ch_type = _picks_by_type(evoked.info)[0][0]
+    if ch_type not in ['mag', 'grad', 'eeg']:
+        raise ValueError("Channel type not supported. Supported channel "
+                         "types include 'mag', 'grad' and 'eeg'.")
+    if times is None:
+        times = np.linspace(evoked.times[0], evoked.times[-1], 10)
+    times = np.array(times)
+
+    if times.ndim != 1:
+        raise ValueError('times must be 1D, got %d dimensions' % times.ndim)
+
+    frames = [np.abs(evoked.times - time).argmin() for time in times]
+
+    blit = False if plt.get_backend() == 'MacOSX' else True
+    picks, pos, merge_grads, _, ch_type = _prepare_topo_plot(evoked,
+                                                             ch_type=ch_type,
+                                                             layout=None)
+    data = evoked.data[picks, :]
+    data *= _handle_default('scalings')[ch_type]
+
+    fig = plt.figure()
+    offset = 0. if blit else 0.4  # blit changes the sizes for some reason
+    ax = plt.axes([0. + offset / 2., 0. + offset / 2., 1. - offset,
+                   1. - offset], xlim=(-1, 1), ylim=(-1, 1))
+    if butterfly:
+        ax_line = plt.axes([0.2, 0.05, 0.6, 0.1], xlim=(evoked.times[0],
+                                                        evoked.times[-1]))
+    else:
+        ax_line = None
+    if isinstance(frames, int):
+        frames = np.linspace(0, len(evoked.times) - 1, frames).astype(int)
+    ax_cbar = plt.axes([0.85, 0.1, 0.05, 0.8])
+    ax_cbar.set_title(_handle_default('units')[ch_type], fontsize=10)
+
+    params = {'data': data, 'pos': pos, 'all_times': evoked.times, 'frame': 0,
+              'frames': frames, 'butterfly': butterfly, 'blit': blit,
+              'pause': False, 'times': times}
+    init_func = partial(_init_anim, ax=ax, ax_cbar=ax_cbar, ax_line=ax_line,
+                        params=params, merge_grads=merge_grads)
+    animate_func = partial(_animate, ax=ax, ax_line=ax_line, params=params)
+    pause_func = partial(_pause_anim, params=params)
+    fig.canvas.mpl_connect('button_press_event', pause_func)
+    key_press_func = partial(_key_press, params=params)
+    fig.canvas.mpl_connect('key_press_event', key_press_func)
+    if frame_rate is None:
+        frame_rate = evoked.info['sfreq'] / 10.
+    interval = 1000 / frame_rate  # interval is in ms
+    anim = animation.FuncAnimation(fig, animate_func, init_func=init_func,
+                                   frames=len(frames), interval=interval,
+                                   blit=blit)
+    fig.mne_animation = anim  # to make sure anim is not garbage collected
+    plt.show()
```

Comment:
this is dangerous ... you need to expose it to the user. Otherwise, all plots which are open but don't use `plt.close()` will open when you call this.